### PR TITLE
Fix tests by running them with old mypy options and add legacy mode

### LINKED ISSUE #32: fix the broken tests
I assume the breaking changes introduced broken tests.

### LINKED ISSUE #98: `--legacy` mode

### PR DESCRIPTION
**Pull request by KotlinIsland**
legacy mode is now among us 😳

closes #32
closes #98

### CODE REVIEW

File: mypy/test/testcheck.py [hunk 1]
```diff
@@ -221,7 +228,8 @@ def run_case_once(self, testcase: DataDrivenTestCase,
             output = testcase.output2.get(incremental_step, [])
         else:
             raise AssertionError()
-
+        # BASED: inject sussy new line
+        output = [it.replace("$NL", "\n") for it in output]
```
Comment (DetachHead) [inline mypy/test/testcheck.py:232]: what this means

File: test-data/unit/check-flags.test [hunk 2]
```diff
@@ -1439,7 +1441,7 @@ def f(c: A) -> None:  # E: Missing type parameters for generic type "A"
     pass
 [file mypy.ini]
 \[mypy]
-strict = True
+disallow_any_generics = True
```
Comment (DetachHead) [inline test-data/unit/check-flags.test:1444]: why replace `strict` with this specific flag

File: tox.ini [hunk 3]
```diff
@@ -51,7 +51,7 @@ description = type check ourselves
 basepython = python3.7
 commands =
     python -m mypy --config-file mypy_self_check.ini -p mypy -p mypyc
-    python -m mypy --config-file mypy_self_check.ini --baseline-file .mypy/proper_plugin.json misc/proper_plugin.py
+    python -m mypy --config-file mypy_self_check.ini --legacy misc/proper_plugin.py
```
Comment (DetachHead) [inline tox.ini:54]: whats `proper_plugin.json` and why was it removed

File: test-data/unit/check-functions.test [hunk 4]
```diff
@@ -2085,7 +2085,7 @@ class A(Generic[t]):
         return None
 [builtins fixtures/bool.pyi]
 [out]
-main:5: error: Cannot use a covariant type variable as a parameter
+main:5: error: This usage of this covariant type variable is unsafe as an input parameter.$NLIf this is intentional and you know what you are doing, you can ignore this line with 'unsafe-variance'
```
Comment (DetachHead) [inline test-data/unit/check-functions.test:2088]: > This usage of this

sounds a bit odd. maybe reword to be like 
> A covariant type variable is unsafe as an input parameter.

File: mypy/test/testcheck.py (hunk 1 above)
Comment (DetachHead) [reply mypy/test/testcheck.py:232]: oh its for errors with newllines in thghem, maybe add a comment saying that? or can you put `\n` directly in the test files? or nah

Comment (KotlinIsland) [reply mypy/test/testcheck.py:232]: There was absolutely no possible way to put it in the test data., I'll add a comment

File: test-data/unit/check-flags.test (hunk 2 above)
Comment (KotlinIsland) [reply test-data/unit/check-flags.test:1444]:
```
[case testStrictInConfigAnyGeneric]
# flags: --config-file tmp/mypy.ini
from typing import TypeVar, Generic
```
I'll add a comment

Comment (KotlinIsland) [reply test-data/unit/check-flags.test:1444]: `-- maybe a little useless because theres no strict but w/e`

File: tox.ini (hunk 3 above)
Comment (KotlinIsland) [reply tox.ini:54]: it was a seperate baseline for a different run, I just made that run use legacy mode instead

File: test-data/unit/check-functions.test (hunk 4 above)
Comment (KotlinIsland) [reply test-data/unit/check-functions.test:2088]: 🙄🙄🙄 out of scope?

**Review by DetachHead — APPROVED** [review]
